src/nvim/buffer.c

```diff
@@ -2833,15 +2816,16 @@ build_stl_str_hl (
       Trunc
     }               type;
   }           item[STL_MAX_ITEM];
-  int minwid;
-  int maxwid;
-  int zeropad;
-  char_u base;
-  char_u opt;
+
+  typedef enum {
+     DECIMAL     = 10,
+     OCTAL       = 8,
+     HEXIDECIMAL = 16
+  } number_base;
```

Comment:
It is very strange to see `typedef` inside the function. Also note that according to the style guide this needs to be

```C
typedef enum {
    kNumBaseDecimal = 10,
    kNumBaseOctal = 8,
    kNumBaseHexadecimal = 16,
} NumberBase;
```
```